src/libraries/System.Linq/src/System/Linq/OrderBy.cs

```diff
@@ -7,12 +7,86 @@ namespace System.Linq
             new OrderedEnumerable<TSource, TKey>(source, keySelector, null, false, null);
 
         public static IOrderedEnumerable<TSource> OrderBy<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector, IComparer<TKey>? comparer) =>
             new OrderedEnumerable<TSource, TKey>(source, keySelector, comparer, false, null);
 
+        /// <summary>
+        /// Sorts the elements of a sequence in descending order.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <param name="source">A sequence of values to order.</param>
+        /// <returns>An <see cref="IOrderedEnumerable{TElement}"/> whose elements are sorted.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// This method is implemented by using deferred execution. The immediate return value is an object
+        /// that stores all the information that is required to perform the action.
+        /// The query represented by this method is not executed until the object is enumerated by calling
+        /// its <see cref="IEnumerable{T}.GetEnumerator"/> method.
+        ///
+        /// This method compares elements by using the default comparer <see cref="Comparer{T}.Default"/>.
+        /// </remarks>
+        public static IOrderedEnumerable<T> OrderDescending<T>(this IEnumerable<T> source) =>
+            OrderByDescending(source, static element => element);
+
+        /// <summary>
+        /// Sorts the elements of a sequence in descending order.
+        /// </summary>
+        /// <typeparam name="T">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <param name="source">A sequence of values to order.</param>
+        /// <param name="comparer">An <see cref="IComparer{T}"/> to compare keys.</param>
+        /// <returns>An <see cref="IOrderedEnumerable{TElement}"/> whose elements are sorted.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// This method is implemented by using deferred execution. The immediate return value is an object
+        /// that stores all the information that is required to perform the action.
+        /// The query represented by this method is not executed until the object is enumerated by calling
+        /// its <see cref="IEnumerable{T}.GetEnumerator"/> method.
+        ///
+        /// If comparer is <see langword="null"/>, the default comparer <see cref="Comparer{T}.Default"/> is used to compare elements.
+        /// </remarks>
+        public static IOrderedEnumerable<T> OrderDescending<T>(this IEnumerable<T> source, IComparer<T> comparer) =>
+            OrderByDescending(source, static element => element, comparer);
```

Comment:
Not very important, but I'm wondering if we could somehow eliminate the null check for the key selector. One way might be to remove the null check from the `OrderedEnumerable` constructor
https://github.com/dotnet/runtime/blob/6a21dbf9ffa5d9fb0adb931b4fad97256e0565ee/src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs#L119-L122
and place it in the methods that do accept it as a parameter.

Reply:
Could be done like this. Aren't such simple null checks where constant null is passed optimized away?

Reply:
Not if the constant is defined in a different method. But again, not very important.

Reply:
was fixed in last commit